pkg/index/index.go

```diff
@@ -204,8 +206,7 @@ type SeriesDocument struct {
 type SeriesStore interface {
 	Store
 	// Search returns a list of series that match the given matchers.
-	Search(context.Context, []SeriesMatcher, []FieldKey) ([]SeriesDocument, error)
-	Execute(context.Context, Query) (posting.List, error)
+	Search(context.Context, []SeriesMatcher, []FieldKey, Query) ([]SeriesDocument, error)
```

Comment:
I might merge `SeriesMatcher` and `Query` into `Query`. The storage.index could build the query first, then pass it to the `Search`.